### PR TITLE
Relicense to MIT

### LINKED ISSUE #1649: Permissive licensing
Discussion with JK Weiss, TESCAN:

If LiberTEM was under a permissive license like MIT, it would allow integration in TESCAN's software. That would be of concrete interest for TESCAN.

Since we have a lot of code contributed by a small number of contributors, and we can probably contact a large percentage of our contributors, re-licensing would be a realistic option.

### PR DESCRIPTION
**Pull request by uellue**
This is built on top of #1687, on which it depends.

* Remove Holo code that was copied from HyperSpy under the GPL
* Relicense LiberTEM under MIT license

We will need to release LiberTEM-holo and handle the migration and deprecation before merging this. Marked as draft for that reason.

It also may require a rebuilt client since `client/src/About.tsx` was changed.
    
Closes #1649

## Contributor Checklist:

* [x] I have added or updated my entry in [the creators.json file](https://github.com/LiberTEM/LiberTEM/blob/master/packaging/creators.json)
* [x] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM/tree/master/docs/source/changelog) for my contribution
* [x] I have added/updated documentation for all user-facing changes
* [x] I have added/updated test cases
* [x] I have included the [rebuilt production build of the client](https://libertem.github.io/LiberTEM/contributing.html?#building-the-client) (only if changes were made to the GUI)

## Reviewer Checklist:

* [ ] `/azp run libertem.libertem-data` passed
* [x] No import of GPL code from MIT code

### CODE REVIEW

**Comment by uellue** [comment]
Discussion with @sk1p: We don't know any users of the previous holo code and development has moved to https://github.com/LiberTEM/LiberTEM-holo. This is currently the recommended implementation even though it is not released yet.

For that reason it seems better overall to not delay the MIT relicensing and remove the GPL holo code with a hard cut.

**Comment by uellue** [comment]
"About" in GUI confirmed to show the new text.

**Comment by uellue** [comment]
The live integration error looks unrelated. Strange, what is going on there?

**Comment by sk1p** [comment]
> The live integration error looks unrelated. Strange, what is going on there?

That was probably because of the overloaded CI server; I've triggered a re-run.

File: docs/source/releasing.rst [hunk 1]
```diff
@@ -60,8 +60,7 @@ When planning a release, create a new issue with the following checklist:
     * [ ] Review open issues and pull requests
     * [ ] Confirm that pull requests and issues are handled as intended, i.e. milestoned and merged
           in appropriate branch.
-    * [ ] License review: no import of GPL code from MIT code
-          `pydeps --only "libertem" --show-deps --noshow src\libertem | python scripts\licensecheck.py`
+    * [ ] License review: compatible with MIT license
```
Comment (sk1p) [inline docs/source/releasing.rst:63]: The script could be removed now, right?

**Review by sk1p — COMMENTED** [review]
Generally LGTM; two small nits

File: docs/source/applications.rst [hunk 2]
```diff
@@ -24,7 +24,6 @@ and also integrated into this documentation via the links below:
    app/amorphous
    app/strain
    app/phasechange
-   app/holography
```
[the remark this answers is not in the document]
Comment (sk1p) [reply docs/source/applications.rst:27]: ... meaning to where the documentation of LiberTEM-holo will live in the future

[the remark this answers is not in the document]
Comment (uellue) [reply docs/source/applications.rst:27]: Discussion with @sk1p : LiberTEM-holo is linked in packages.rst that is referenced in the paragraphs above. Restructuring this properly is a larger TODO for later (tm).

File: docs/source/releasing.rst (hunk 1 above)
Comment (uellue) [reply docs/source/releasing.rst:63]: Good point, thx!

**Pull request merged**